core/src/test/scala/unit/kafka/controller/ControllerIntegrationTest.scala

```diff
@@ -572,6 +572,48 @@ class ControllerIntegrationTest extends QuorumTestHarness {
       "failed to remove reassign partitions path after completion")
   }
 
+  @Test
+  def testAutoPreferredReplicaLeaderElectionFollowedByAnotherNewReassignment(): Unit = {
+    servers = makeServers(4, autoLeaderRebalanceEnable = true)
+    val controller = getController().kafkaController
+    val tp = new TopicPartition("t", 0)
+    val assignment = Map(tp.partition -> Seq(0, 2))
+    TestUtils.createTopic(zkClient, tp.topic, partitionReplicaAssignment = assignment, servers = servers)
+
+    // Shutdown broker 2 and reassign partition tp from [0, 2] to [1, 0, 2] to create a stuck reassignment.
```

Comment:
In a typical reassignment, we don't change the replication factor. Could we design the test to change replicas from (0, 2) to (1, 0)? Ditto below.

Reply:
In order to reproduce this issue, broker (0,1) should be always in the replica set. Maybe it's easier to start with replicas (0, 2, 4) instead of (0, 2), so that the replication factor is always 3. Code has been updated. Could you help to take a look? Thanks.